src/lib/pipes/normalise-schedule-pipe.js

```diff
@@ -0,0 +1,263 @@
+    }
+    if(schemaDay == "https://schema.org/Saturday" || schemaDay == "SA"){
+        return RRule.RRule.SA;
+    }
+    if(schemaDay == "https://schema.org/Sunday" || schemaDay == "SU"){
+        return RRule.RRule.SU;
+    }
+  }
+
+  makeDate(dateString, timeString){
+    let fullString = dateString;
+    if (typeof timeString !== 'undefined'){
+        fullString = fullString + "T" + timeString;
+    }
+    return new Date(fullString);
+  }
+
+  byWeekDay(schedule){
+    if (typeof schedule.byDay !== 'undefined'){
+        let weekDays = [];
+        for(let day of schedule.byDay){
+            weekDays.push(this.schemaDaytoRRuleDay(day));
+        }
+        return weekDays;
+    }
+  }
+
+  byMonth(schedule){
+    return schedule.byMonth;
+  }
+
+  byMonthDay(schedule){
+    return schedule.byMonthDay;
+  }
+
+  dtStart(schedule){
+    return this.makeDate(schedule.startDate, schedule.startTime);
+  }
+
+  until(schedule){
+    if(typeof schedule.endDate !== 'undefined'){
+        return this.makeDate(schedule.endDate, schedule.endTime);
+    }
+  }
+
+  freq(schedule){
+    try{
+        let frequency = parseDuration(schedule.repeatFrequency);
+        // Making some sweeping assumptions about value here,
+        // eg. that only one of the periods will be set: only day, or only month, etc
+        // Doesn't bother with minutes or seconds
+        // Returns {frequency, interval}
+        // Returns interval 1 by default except in special cases
+        if(frequency.hours == 1){
+            return {freq: RRule.RRule.HOURLY, interval: 1};
+        }
+        if(frequency.days == 1 || frequency.hours == 24){
+            return {freq: RRule.RRule.DAILY, interval: 1};
+        }
+        if(frequency.weeks == 1 || frequency.days == 7){
+            return {freq: RRule.RRule.WEEKLY, interval: 1};
+        }
+        if(frequency.weeks == 2 || frequency.days == 14){
+            return {freq: RRule.RRule.WEEKLY, interval: 2};
+        }
+        if(frequency.months == 1){
+            return {freq: RRule.RRule.MONTHLY, interval: 1};
+        }
+        if(frequency.years == 1 || frequency.months == 12){
+            return {freq: RRule.RRule.YEARLY, interval: 1};
+        }
+    }catch(error){
+        console.log(`Could not parse duration [${schedule.repeatFrequency}]`);
+        console.log(`Therefore guessing repeat frequency from byDay..`);
+        // console.log(error);
+
+        if(typeof schedule.byDay !== 'undefined' && schedule.byDay.length > 0){
+            // This is a hack to accommodate a common non-conformant practice of using byDay
+            // to indicate weekly repetition.
+            return {freq: RRule.RRule.WEEKLY, interval: 1};
+        }
+    }
+  }
+
+  count(schedule){
+    return schedule.repeatCount;
+  }
+
+  eventsUntil(until){
+    // Generate events until two weeks from now, or until the end date,
+    // whichever is sooner.
+    let twoWeeks = new Date(Date.now() + 12096e5);
```

Comment:
Can this be a new setting variable? I've tried to make every magic number be put in there for ease of code readability and changing later.